src/server/routes/mkurl/mkurl.ts

```diff
@@ -16,89 +16,45 @@
-  return deserialize(serialize(appSettings));
-}
-
-function convertToClientDataCube(cube: QueryableDataCube): ClientDataCube {
-  return {
-    ...cube,
-    timeAttribute: cube.timeAttribute && cube.timeAttribute.name
-  };
-}
-
-export function mkurlRouter(settings: Pick<SettingsManager, "getSources" | "appSettings">) {
+export function mkurlRouter(settings: Pick<SettingsManager, "getSources" | "appSettings" | "logger">) {
 
   const router = Router();
 
   router.post("/", async (req: Request, res: Response) => {
-    const { dataCubeName, viewDefinitionVersion, viewDefinition } = req.body;
-
-    if (typeof viewDefinitionVersion !== "string") {
-      res.status(400).send({ error: "must have a viewDefinitionVersion" });
-      return;
-    }
 
-    const definitionConverter = definitionConverters[viewDefinitionVersion as ViewDefinitionVersion];
-
-    if (definitionConverter == null) {
-      res.status(400).send({ error: "unsupported viewDefinitionVersion value"
-      });
-      return;
-    }
-
-    if (typeof dataCubeName !== "string") {
-      res.status(400).send({ error: "must have a dataCubeName" });
-      return;
-    }
-
-    if (typeof viewDefinition !== "object") {
-      res.status(400).send({ error: "viewDefinition must be an object" });
-      return;
-    }
-
-    let sources: Sources;
     try {
-      sources = await settings.getSources();
-    } catch (e) {
-      res.status(400).send({ error: "Couldn't load settings" });
-      return;
-    }
-    const myDataCube = getDataCube(sources, dataCubeName);
-    if (!myDataCube) {
-      res.status(400).send({ error: "unknown data cube" });
-      return;
-    }
+      const dataCube = await parseDataCube(req, settings);
+      const viewDefinition = parseViewDefinition(req);
+      const converter = parseViewDefinitionConverter(req);
 
-    if (!isQueryable(myDataCube)) {
-      res.status(400).send({ error: "un queryable data cube" });
-      return;
-    }
+      const essence = createEssence(viewDefinition, converter, dataCube, settings.appSettings);
 
-    const clientDataCube = convertToClientDataCube(myDataCube);
-    const clientAppSettings = convertToClientAppSettings(settings.appSettings);
+      const hash = `#${dataCube.name}/${urlHashConverter.toHash(essence)}`;
+      res.json({ hash });
+    } catch (error) {
+      if (isValidationError(error)) {
+        res.status(error.code).send({ error: error.message });
+        return;
+      }
 
-    let essence: Essence;
-    try {
-      essence = definitionConverter.fromViewDefinition(viewDefinition, clientAppSettings, clientDataCube);
-    } catch ({ message }) {
-      res.status(400).send({ error: "invalid viewDefinition object", message });
-      return;
-    }
+      settings.logger.error(errorToMessage(error));
 
-    res.json({
-      hash: `#${myDataCube.name}/${urlHashConverter.toHash(essence)}`
-    });
+      res.status(500).send({
+        error: "could not compute",
```

Comment:
Could you explain what you mean? I don't understand, probably because you use word "message" here for two different things and in code it is another one.